language/compiler/bytecode_source_map/src/disassembler.rs

```diff
@@ -132,11 +224,219 @@ impl<Location: Clone + Eq + Default> Disassembler<Location> {
             .zip(kinds.iter())
             .map(|((name, _), kind)| format!("{}: {:#?}", name.as_str(), kind))
             .collect();
-        if ty_params.is_empty() {
-            "".to_string()
-        } else {
-            format!("<{}>", ty_params.join(", "))
+        Self::enclose_ty_params(ty_params)
+    }
+
+    fn disassemble_bytecode(
+        &self,
+        function_definition_index: FunctionDefinitionIndex,
+    ) -> Result<String> {
+        if !self.options.print_code {
+            return Ok("".to_string());
         }
+
+        let function_def = self.get_function_def(function_definition_index)?;
+        let locals_sigs = self
+            .source_mapper
+            .bytecode
+            .locals_signature_at(function_def.code.locals);
+        let function_source_map = self
+            .source_mapper
+            .source_map
+            .get_function_source_map(function_definition_index)?;
+
+        let name_for_local = |local_idx| -> Result<(String, String)> {
+            let name = function_source_map
+                .get_local_name(local_idx)
+                .ok_or_else(|| {
+                    format_err!(
+                        "Unable to get local name at index {} while disassembling location-based instruction", local_idx
+                    )
+                })?
+                .0;
+            let sig_tok = locals_sigs.0.get(local_idx as usize).ok_or_else(|| {
+                format_err!("Unable to get type for local at index {}", local_idx)
+            })?;
+            let ty =
+                self.disassemble_sig_tok(sig_tok.clone(), &function_source_map.type_parameters)?;
+            Ok((name.to_string(), ty))
+        };
+
+        let mut instrs: Vec<String> = function_def
+            .code
+            .code
+            .iter()
+            .enumerate()
+            .map(|(instr_index, instruction)| match instruction {
```

Comment:
`instr_index` and indentation are now formatted by each individual instruction. Is it possible we do it in an outer loop/map? I think having a single place controlling these will be more consistent and readable.